sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClient.java

```diff
@@ -133,17 +139,17 @@ public Mono<HttpResponse> send(HttpRequest request, Context context) {
                     return Mono.just(response);
                 }
             })
-            .onErrorMap(throwable -> {
+            .onErrorResume(throwable -> {
                 // The exception was an SSLException that was caused by a failure to connect to a proxy.
                 // Extract the inner ProxyConnectException and propagate that instead.
-                if (throwable instanceof SSLException && throwable.getCause() instanceof ProxyConnectException) {
-                    return throwable.getCause();
+                if (!proxyRetry && addProxyHandler && throwable instanceof ProxyConnectException
+                    || (throwable instanceof SSLException && throwable.getCause() instanceof ProxyConnectException)) {
+                    return attemptAsync(request, eagerlyReadResponse, ignoreResponseBody, headersEagerlyConverted,
+                        responseTimeout, progressReporter, true);
```

Comment:
IIUC, for the `ProxyConnectException` in `SSLException` condition, the retry cycle repeats as long as the condition is true (could be forever).  Earlier there seems to be a max limit of 1 applied, is it intentional that we removed the upper limit? Should !proxyRetry applied for both conditions.

Reply:
Good catch this was meant to be

```
!proxyRetry 
    && (throwable instanceof ProxyConnectException || (throwable instanceof SSLException && throwable.getCause() instanceof ProxyConnectException)
```

I'll add a test for this as well to make sure this doesn't spiral out of control